### PR TITLE
fix callback err type should include null

### PR DESCRIPTION
**Pull request by luffy84217**
### Describe Bug
See #15029 

### Current problem
The type of err in the callback of most functions in /lib are `Error | undefined` or `WebpackError | undefined`. If the compilation is successful, the actual value of err is `null`.

### Solution
Compare the callback of functions in /lib, and find out those which should be corrected to `Error | null` or `WebpackError | null` then run `special-lint-fix` to update [`type.d.ts`](https://github.com/webpack/webpack/blob/main/types.d.ts).

**What kind of change does this PR introduce?**
type declaration fix

**Did you add tests for your changes?**
No tests were added but the changes pass `typings-test`

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

**Comment by alexander-akait** [comment]
Looks good, please run `yarn special-lint-fix`

**Comment by luffy84217** [comment]
@alexander-akait I run `yarn special-lint-fix` locally to update `type.d.ts` and include it to commit but It fails on azure pipeline job `Run linting` since the job doesn't run `yarn special-lint-fix` to update `type.d.ts`. How to solve this problem? Sorry, this is my first time contributing to webpack.

**Comment by alexander-akait** [comment]
Based on output we have `types.d.ts need to be updated.`, can you check it locally? What is different?

**Comment by luffy84217** [comment]
@alexander-akait Based on azure output it only run `special-lint`, no `special-lint-fix`, that's why `we have types.d.ts need to be updated.`
here's the difference of `type.d.ts` https://github.com/webpack/webpack/pull/15050/files#diff-a150fb53afee945f23f3fe7c176007111f3ba81bb0d188af4077d4a0fc4caa37

**Comment by luffy84217** [comment]
PR #15049 checks also got the same problem on azure pipeline job `Run lint`

**Comment by luffy84217** [comment]
@alexander-akait is it possible to create a new branch and modify workflow job `lint` (e.g. replace `special-lint` to `special-lint-fix`) or if it exists, so that I could be able to merge commit into that branch from [`luffy84217:fix-callback-err-type-should-include-null`](https://github.com/luffy84217/webpack/tree/fix-callback-err-type-should-include-null) and make checks successful.

File: types.d.ts [hunk 1]
```diff
@@ -7197,7 +7220,7 @@ declare abstract class MultiWatching {
 	invalidate(callback?: any): void;
 	suspend(): void;
 	resume(): void;
-	close(callback: CallbackFunction<void>): void;
+	close(callback: CallbackFunction_1<void>): void;
```
Comment (sokra) [inline types.d.ts:7223]: You missed a few places:

* MultiWatching.close
* Watching.close
* Watching.invalidate
* Watching.handler
* Watching.callbacks
* AsyncQueue.add
* AsyncQueue.waitFor

**Review by sokra — CHANGES_REQUESTED** [review]

File: types.d.ts (hunk 1 above)
Comment (luffy84217) [reply types.d.ts]: I commit ([0105d7d](https://github.com/luffy84217/webpack/commit/0105d7d3e6a2d1a0bb18b7cd9fdf83ec78121cc7)) to make up omitted callback types you listed

**Comment by alexander-akait** [comment]
@luffy84217 Can you regenerate types?

**Comment by luffy84217** [comment]
@alexander-akait Yes, I did, check this change ([0105d7#diff-a150fb5](https://github.com/webpack/webpack/pull/15050/commits/0105d7d3e6a2d1a0bb18b7cd9fdf83ec78121cc7#diff-a150fb53afee945f23f3fe7c176007111f3ba81bb0d188af4077d4a0fc4caa37))

**Comment by alexander-akait** [comment]
Can you run `yarn special-lint-fix`, again?

**Comment by luffy84217** [comment]
run `yarn special-lint-fix` locally and commit again?

**Comment by alexander-akait** [comment]
Run `yarn lint` locally, what is output, maybe you forget to push?

**Comment by luffy84217** [comment]
@alexander-akait 
$ yarn lint
yarn run v1.22.17
$ yarn setup
$ node ./setup/setup.js
$ yarn code-lint && yarn special-lint && yarn type-lint && yarn typings-test && yarn module-typings-test && yarn yarn-lint && yarn pretty-lint && yarn spellcheck
$ eslint . --ext '.js' --cache
$ node node_modules/tooling/lockfile-lint && node node_modules/tooling/schemas-lint && node node_modules/tooling/inherit-types && node node_modules/tooling/format-schemas && node tooling/generate-runtime-code.js && node tooling/generate-wasm-code.js && node node_modules/tooling/format-file-header && node node_modules/tooling/compile-to-definitions && node node_modules/tooling/precompile-schemas && node node_modules/tooling/generate-types --no-template-literals
Last converge was suboptimial.
$ tsc
$ tsc -p tsconfig.types.test.json
$ tsc -p tsconfig.module.test.json
$ yarn-deduplicate --fail --list -s highest yarn.lock
$ yarn pretty-lint-base --check
$ prettier "*.{ts,json,yml,yaml,md}" "{setup,lib,bin,hot,benchmark,tooling,schemas}/**/*.json" "examples/*.md" --check
Checking formatting...
All matched files use Prettier code style!
$ cspell "{.github,benchmark,bin,examples,hot,lib,schemas,setup,tooling}/**/*.{md,yml,yaml,js,json}" "*.md"
CSpell: Files checked: 1072, Issues found: 0 in 0 files
Done in 103.36s.

**Comment by alexander-akait** [comment]
Very weird, because running yarn link on CI output the error, double check your git

**Comment by luffy84217** [comment]
@alexander-akait Problem is solved. My device ignores comment formatting. See [f0a8664](https://github.com/webpack/webpack/pull/15050/commits/f0a8664895746ebcd7a35e5947c54fe0314210e5)

**Review by sokra — APPROVED** [review]

**Comment by sokra** [comment]
Thanks

**Pull request merged**